### ISSUE TITLE
Pose and pose_inv differ significantly

### ISSUE THREAD

**Issue by brandonwagstaff** [JiawangBian/SC-SfMLearner-Release#46]
I've noticed that the pose estimates, and their inverse (computed through `compute_pose_with_inv`) are consistently different from each other. It turns out that the predicted 'forward' translation (the third element of the pose prediction)  is always larger when moving 'forward' in time, and smaller when moving 'backward' in time. That is, the predicted pose change from image t to t+1 is consistently  larger than the predicted pose change from t+1 to t. You can reproduce this by adding the following lines to the `train()` or `validate_without_gt()` functions in `train.py` , and observing their values for a pretained model (I used the updated resnet18_depth_256 model) when training/evaluating the model:

`print('forward', poses[0][:,2].mean().item(), poses_inv[0][:,2].mean().item())
print('backward', poses[1][:,2].mean().item(), poses_inv[1][:,2].mean().item())`

Can you comment on why this might be the case? I would expect that between two images, the predicted forward translation should be the same magnitude, with one being positive and one being negative.

**Comment by JiawangBian** [comment]
Hi, 

Thanks for your question. For reproducing, I do it in test_vo.py (kitti sequence 09 using resnet18_depth_256 model.). There is no data augmentation as in training. so it would be more clear. I debug by adding following codes, and results show that the t3-error between forward and backward is about 2e-3. Does it match your finding?

![Screenshot from 2020-05-07 11-29-37](https://user-images.githubusercontent.com/11647217/81246238-29bc1280-9056-11ea-8675-a1404eef8ce3.png)

**Comment by brandonwagstaff** [comment]
Yes, that matches what I have. There is a consistent difference in translation of about 10-25% for all samples, which seems like an issue. I was wondering if you had any insight as to why this would occur - possibly there is some mechanism that causes the loss function to increase if more 'black' space exists on the reconstructed images?

**Comment by JiawangBian** [comment]
It should be consistent, but it is very difficult for the network to predict very consistent results. If you find this issue important, you may minimize this error during training as  a loss function. I did this before, and the loss looks small and impact little for the training.
